src/qml/main.qml

```diff
@@ -122,58 +127,89 @@ ApplicationWindow {
         onOpenLogs: Qt.openUrlExternally(CSLOLUtils.toFile("./log.txt"))
     }
 
-    onClosing: {
-        if (cslolTrayIcon.available && settings.enableSystray) {
+    onClosing: function(close) {
+        if (systemTrayIcon && settings.enableSystray) {
             close.accepted = false
             window.hide()
+            systemTrayIcon.updateWindowVisibility(false)
         }
     }
 
-    SystemTrayIcon {
-        id: cslolTrayIcon
-        visible: settings.enableSystray
-        iconSource: "qrc:/icon.png"
-        tooltip: "cslol-manager"
-        menu: Menu {
-            MenuItem {
-                text: !window.visible ? qsTr("Show") : qsTr("Minimize")
-                onTriggered: window.visible ? window.hide() : window.show()
-            }
-            MenuItem {
-                text: window.patcherRunning ? qsTr("Stop") : qsTr("Run")
-                onTriggered: {
-                    if (window.patcherRunning) {
-                        cslolToolBar.stopProfile()
-                    } else if (!window.isBussy) {
-                        cslolToolBar.saveProfileAndRun(true)
-                    }
+    function createSystemTrayIcon() {
+        if (!systemTrayIcon) {
+            var component = Qt.createComponent("CSLOLSystemTrayIcon.qml");
```

Comment:
Do we really have to resort to dynamically creating  this?

Reply:
I initially went with a visibility based approach, but I discovered that when I toggled the Enable systray switch multiple times, the context menu would stop working altogether, while the icon still appeared. Maybe I was missing something obvious (this iteration of the program is commit ac90a9e0b1ee652ffdeb439b236501cad46fdcea), but with my previous problems with the application window visibility, I figured I could just do it this way and sidestep that altogether.